clang/lib/Basic/LangStandards.cpp

```diff
@@ -10,9 +10,49 @@
 #include "clang/Config/config.h"
 #include "llvm/ADT/StringSwitch.h"
 #include "llvm/Support/ErrorHandling.h"
+#include "llvm/Support/FormatVariadic.h"
 #include "llvm/TargetParser/Triple.h"
 using namespace clang;
 
+const char *clang::languageToString(Language L) {
+  // I would like to make this function and the definition of Language
+  // in the .h file simply expand the contents of a .def file.
+  // However, in the .h the members of the enum have doxygen annotations
+  // and/or comments which would be lost.
```

Comment:
I'm not sure I have a good solution for this, but I'd like to see comments on this from the wider clang community.  Maybe it would make sense to submit just this function as a separate PR.

Reply:
Sure, I can create another PR for this.